core/src/test/java/org/infinispan/manager/CacheManagerTest.java

```diff
@@ -616,4 +611,23 @@ public void stop() {
          stopped = true;
       }
    }
+
+   class LatchGlobalMarshaller extends GlobalMarshaller {
+      private final CompletableFuture<Void> managerStopBlocked;
+      private final CompletableFuture<Void> managerStopResumed;
+
+      public LatchGlobalMarshaller(CompletableFuture<Void> managerStopBlocked,
+                                   CompletableFuture<Void> managerStopResumed) {
+         this.managerStopBlocked = managerStopBlocked;
+         this.managerStopResumed = managerStopResumed;
+      }
+
+      @Override
+      public void stop() {
+         log.tracef("Stopping global component registry");
+         managerStopBlocked.complete(null);
+         managerStopResumed.join();
```

Comment:
one is one complete and the other join ?

Reply:
Yes, it's moved from https://github.com/infinispan/infinispan/pull/6978/files/e2737cb7482462a0e67fbdc1d561697163d8be67#diff-1ed883ada0e6bc47a12ece1101e8b891L280